MentorsManagement.API/Services/MentorService.cs

```
@@ -1,62 +1,59 @@
-﻿using MentorsManagement.API.DBContexts;
+﻿using MentorsManagement.API.Data;
 using MentorsManagement.API.Models;
-using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using MongoDB.Driver;
 
 namespace MentorsManagement.API.Services
 {
     public class MentorService : IMentorService
     {
-        private readonly IMentorDbContext _context;
-
-        public MentorService(IMentorDbContext context)
+        private readonly IMongoCollection<Mentor> _mentors;
+        public MentorService(IMongoDatabase mongoDatabase, IOptions<MongoDbSettings> mongoDbSettings)
         {
-            _context = context;
+            _mentors = mongoDatabase.GetCollection<Mentor>(mongoDbSettings.Value.MentorsCollectionName);
         }
 
         public async Task<List<Mentor>> GetAllMentors()
         {
-            return await _context.Mentors.ToListAsync();
+            var allMentors = await _mentors.Find(_ => true).ToListAsync();
+            return allMentors;
         }
-
-        public async Task<Mentor?> GetMentorById(int id)
+        public async Task<Mentor?> GetMentorById(string id)
         {
-            return await _context.Mentors.FindAsync(id);
+            var mentor = await _mentors.Find(m => m.Id == id).FirstOrDefaultAsync();
+            return mentor;
         }
 
         public async Task<Mentor?> CreateMentor(Mentor mentor)
         {
-            var isMentorExisting = await _context.Mentors.FindAsync(mentor.MentorId);
-            if (isMentorExisting != null)
+            if (mentor.Id != string.Empty)
             {
                 return null;
```

this should throw an error